lib/mixins/properties-changed.html

```diff
@@ -332,13 +332,31 @@
          * @protected
          */
         _flushProperties() {
-          if (this.__dataPending && this.__dataOld) {
-            let changedProps = this.__dataPending;
+          const props = this.__data;
+          const changedProps = this.__dataPending;
+          const old = this.__dataOld;
+          if (this._shouldPropertiesChange(props, changedProps, old)) {
             this.__dataPending = null;
```

Comment:
Just a thought re: memory: should we be zeroing out `__dataOld` here, to so that the previous turn's worth of changed data can be gc'ed?  It currently gets reset on first change in new turn [here](https://github.com/Polymer/polymer/pull/5097/files#diff-980fcce29657756129c5008e85181523R271).